Add stock-location-pi endpoint

parisiale opened the pull request.

parisiale committed:
Add specs for StockLocation PI endpoint

Updating README and adding JSON schema for the PI request.

parisiale committed:
Add stock-location-pi-count POST endpoint

parisiale committed:
Add trivial tests for PI request schema validation

parisiale commented on controllers/stock_location.py:
```diff
@@ -1,14 +1,41 @@
+import json
+import os
+
+import jsonschema
```
New dependency.

parisiale committed:
Don't do schema validation for consistency

Here we avoid validating PI requests by using JSON
schema validation for consistency with other
endpoints.

parisiale committed:
Fix location check in PI endpoint

parisiale committed:
Fix endpoint definition

parisiale committed:
Update PI request schema: package_name required for adj

parisiale committed:
Rename PI endpoint param to pi_request

Using `pi_request` instead of `request` to avoid clashing
with odoo internals during endpoint definition.

ws-mps commented on README.md:
````diff
@@ -334,6 +335,16 @@ Params:
+Params:
+@param pi_request: JSON object with "pi_count_moves", "inventory_adjustments", "preceding_inventory_adjustments" and "location_id" entries
+```
+
````
Can we add a brief description of what it is going to do?

parisiale replied:
Done.

ws-mps commented on controllers/stock_location.py:
```diff
@@ -27,11 +29,45 @@ def get_location(self, location_id=None, location_name=None, location_barcode=No
+            raise ValidationError(
+                _('You need to provide a valid id for the location.'))
+
+        location = Location.browse(location_id)
```
We can use `Location.get_location(location_id)`, then lines 70 and 71 are not needed.

parisiale committed:
PI request schema update

Adding optional 'lot_name' entry for inventory adjustments.
Removing constraints on request entries.

parisiale committed:
Use get_location to validate and retrieve locations

parisiale committed:
Improve documentation of PI endpoint

parisiale committed:
Fix entry name in PI endpoint schema

parisiale committed:
Use quant_ids in PI endpoint scheama

We were wrongly using `quants_ids` before.

parisiale committed:
Override udes_core's StockLocation PI method

Setting the warehouse picking type id in the parent method,
so that PI move pickings can be created with the current
warehouse picking type.

parisiale committed:
Update README with new fileds in warehouse info response

We now return PI fields; adding them to the spec.

ws-mps commented on models/stock_location.py:
```diff
@@ -19,3 +19,18 @@ def _prepare_info(self, extended=False, **kwargs):
+        Users = self.env['res.users']
+
+        warehouse = Users.get_user_warehouse()
+        picking_type_id = warehouse.pick_type_id.id
```
This is not the picking type you are looking for

parisiale committed:
Fix the picking type for PI count moves

Here we correctly use the value of the
u_pi_count_move_picking_type for the current user's warehouse
as the picking_type for PI count moves.

parisiale committed:
Fix picking type id retrieval

parisiale committed:
[maint] Remove unused var from package swap test
